include/miopen/miopen.h

```diff
@@ -4704,6 +4706,231 @@ miopenReduceTensor(miopenHandle_t handle,
 /** @} */
 // CLOSEOUT TensorReduce DOXYGEN GROUP
 
+// Find 2.0 API
+/** @addtogroup find2
+ *
+ *  @{
+ */
+
+/*! @brief Describes a problem for different miopen operations.
+ *
+ * For now, this is only used for convolution, but could be used for other
+ * operators in the future(such as GEMM, Pooling, etc)
+ */
+MIOPEN_DECLARE_OBJECT(miopenProblem);
+
+/*! @enum miopenProblemDirection_t
+ * Directions of miopen operation.
+ */
+typedef enum
+{
+    miopenProblemDirectionForward         = 0,
+    miopenProblemDirectionBackward        = 1,
+    miopenProblemDirectionBackwardWeights = 2,
+} miopenProblemDirection_t;
+
+/*! @enum miopenTensorArgumentId_t
+ * Identifiers for tensor arguments of problems and operations.
+ */
+typedef enum
+{
+    miopenTensorArgumentIdInvalid = 0,
+    miopenTensorConvolutionX      = 1,
+    miopenTensorConvolutionW      = 2,
+    miopenTensorConvolutionY      = 3,
+} miopenTensorArgumentId_t;
+
+/*! @enum miopenTensorArgumentId_t
+ * Different ways to sort results of the find call.
+ */
+typedef enum
+{
+    miopenFindResultsOrderByTime   = 0,
+    miopenFindResultsOrderByMemory = 1,
+} miopenFindResultsOrder_t;
+
+/*! @brief Initializes a problem object describing a convolution operation.
+ *
+ * @param problem      Pointer to the problem to initialize
+ * @param operatorDesc Descriptor of the operator to be used
+ * @param direction    Direction of the operation
+ * @return             miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t miopenCreateConvProblem(miopenProblem_t* problem,
+                                                     miopenConvolutionDescriptor_t operatorDesc,
+                                                     miopenProblemDirection_t direction);
+
+/*! @brief Destroys a problem object.
+ *
+ * @param problem Problem to destroy
+ * @return        miopenStatus_t
+ */
+miopenStatus_t miopenDestroyProblem(miopenProblem_t problem);
```

Comment:
Does it have corresponding `miopenCreateProblem` ?

Reply:
`miopenSolution_t` has API for destroy, does it have one for `create` ?

Reply:
It is allocated by miopen `miopenFindSolutions` and `miopenLoadSolution`.

Reply:
Basically, it can't be created and empty at the same time.